test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/EmailPasswordAuthTests.kt

```diff
@@ -0,0 +1,552 @@
+package io.realm.test.mongodb.shared
+
+import io.realm.mongodb.AppException
+import io.realm.mongodb.Credentials
+import io.realm.test.mongodb.TestApp
+import io.realm.test.mongodb.asTestApp
+import io.realm.test.util.TestHelper
+import kotlinx.coroutines.runBlocking
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertFailsWith
+import kotlin.test.assertNotNull
+import kotlin.test.fail
+
+class EmailPasswordAuthTests {
+
+    private lateinit var app: TestApp
+
+    @BeforeTest
+    fun setup() {
+        app = TestApp()
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (this::app.isInitialized) {
+            app.asTestApp.close()
+        }
+    }
+
+    @Test
+    fun registerUser() {
+        runBlocking {
```

Comment:
You could use expression body for this

Reply:
The reason why I didn't convert it to an expression body is because our custom `runBlocking` function returns a `T` produced inside the lambda block and if you use it in a void function (which is the case in these tests) turns out you need to manually add `Unit` as the last expression inside the lambda, which is kind of annoying. That's why I left it as is. I suspect I forgot to write this explanation 🙈